bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/core/QuickTextSearcher.java

```diff
@@ -121,7 +123,27 @@ private SearchInFilesWalker createWalker(PriorityFunction priorities) {
 		job.schedule();
 		return job;
 	}
-
+	/**
+	 * Try to get a content as String. Avoids Streaming.
+	 */
+	private static String toShortString(IFile file) {
+		/**
+		 * Just any number such that the most source files will fit in. And not too
+		 * big to avoid out of memory.
+		 **/
+		int MAX_BUFFER_LENGTH = 999_999; // max 2MB.
```

Comment:
2 MB are `2_097_152` bytes so comment is confusing.... beside that constants should be defined in top of class file, computed values are often easier to read and change using a computation if they are computed e.g.

```suggestion
		int MAX_BUFFER_LENGTH = 2* 1024*1024; //Read 2 MB at maximum.
```

Reply:
The file is typically utf8 but the resulting String is 16bit/char in memory if there is a single non ascii char.
BTW that is all copy&paste from org.eclipse.search.internal.core.text.FileCharSequenceProvider.toShortString(IFile) and just works

Reply:
> BTW that is all copy&paste from 

There it is a static constant on top of the file

>  and just works

Copy & Paste does not proof correctness... the length check will simply never be true (or the implementation is wrong and violates the API contract) as it can under no circumstances return an array with a length larger than requested.

That means the method will never return `null` (unless IO or runtime errors) and therefore your code that checks for `null` here (**in a different context than where it was copied from**!) will never get null and therefore always only read 1 MB even if the file is 1 GB large... so something seems fishy here.